build.cake

```diff
@@ -622,6 +623,7 @@ Task("Publish-DockerHub")
     {
         DockerPush("linux", "dotnetcore", parameters);
         DockerPush("linux", "fullfx", parameters);
+        DockerPush("linux", "centos", parameters);
```

Comment:
Please excuse my ignorance, but wow is this going to affect the naming on [DockerHub](https://hub.docker.com/u/gittools)? I don't quite understand where the existing names repositories [`gitversion-fullfx`](https://hub.docker.com/r/gittools/gitversion-fullfx), [`gitversion-dotnetcore`](https://hub.docker.com/r/gittools/gitversion-dotnetcore), [`gitversion`](https://hub.docker.com/r/gittools/gitversion) and [`libgit2sharp-mono`](https://hub.docker.com/r/gittools/libgit2sharp-mono) come from and how they relate to these `DockerPush()` statements. Will `centos` become a new `tag` in the `gitversion-dotnetcore` repository?

Reply:
> @asbjornu  [`gitversion`](https://hub.docker.com/r/gittools/gitversion) and [`libgit2sharp-mono`](https://hub.docker.com/r/gittools/libgit2sharp-mono) are already obsolete, only fullfx and dotnetcore are maintained. 

As for centos, that is right, the current changes will not create a tag under dotnetcore - good catch, we'll need to adjust the scripts to allow several distros of linux under dotnetcore and fullfx

Reply:
this is the method where the tags are created https://github.com/GitTools/GitVersion/blob/master/build/utils.cake#L174

Reply:
I see. I find it a bit unfortunate that we have 4 different repositories for variations of the same executable. Can we get rid of every repository that isn't called simply `gitversion` and follow [NGINX](https://hub.docker.com/_/nginx?tab=tags)' naming scheme for labels to distinguish different versions of the image? That would give us one repository with tags following the scheme `{version}-{platform}-{framework}`, for example:

- `4.0.0-centos`
- `4.0.1-beta2-debian-fx`
- `4.0.0-beta15-windows-dotnetcore`

I know doing anything about this scheme is outside the scope of this PR, I just want to have a discussion on this so I know which options we have.

Reply:
I suggest creating a separate issue, I can then implement it, and after that we can adjust this PR to the new scheme.

Reply:
@asbjornu please create an issue with the proposed scheme and we can discuss it there

Reply:
Sure, I've created #1572 for further discussion.